analyzers/src/SonarAnalyzer.CSharp/Rules/InterfaceMethodsShouldBeCallableByChildTypes.cs

```diff
@@ -74,65 +71,54 @@ protected override void Initialize(SonarAnalysisContext context)
                 return;
             }
 
-            var classDeclaration = memberDeclaration.FirstAncestorOrSelf<ClassDeclarationSyntax>();
-            if (classDeclaration == null ||
-                classDeclaration.Identifier.IsMissing ||
-                !IsClassTracked(classDeclaration, analysisContext.SemanticModel))
+            var declaration = (TypeDeclarationSyntax)memberDeclaration.FirstAncestorOrSelf<SyntaxNode>(node => node is TypeDeclarationSyntax);
+            if (declaration == null
+                || declaration.Identifier.IsMissing
+                || !IsDeclarationTracked(declaration, analysisContext.SemanticModel))
             {
                 return;
             }
 
-            var hasPublicEquivalentMethod = classDeclaration.Members
-                .OfType<TMemberSyntax>()
-                .Any(member => areMembersEquivalent(member, memberDeclaration));
+            var hasPublicEquivalentMethod = declaration.Members
+                                                       .OfType<TMemberSyntax>()
+                                                       .Any(member => areMembersEquivalent(member, memberDeclaration));
             if (!hasPublicEquivalentMethod)
             {
                 var identifierName = getIdentifierName(memberDeclaration);
 
-                analysisContext.ReportDiagnosticWhenActive(Diagnostic.Create(rule, identifierName.GetLocation(),
-                    classDeclaration.Identifier.ValueText,
-                    string.Concat(explicitInterfaceSpecifier.Name, ".", identifierName.ValueText)));
+                analysisContext.ReportDiagnosticWhenActive(Diagnostic.Create(Rule,
+                                                                             identifierName.GetLocation(),
+                                                                             declaration.Identifier.ValueText,
+                                                                             string.Concat(explicitInterfaceSpecifier.Name, ".", identifierName.ValueText)));
             }
         }
 
-        private static bool IsClassTracked(ClassDeclarationSyntax classDeclaration, SemanticModel semanticModel)
+        private static bool IsDeclarationTracked(BaseTypeDeclarationSyntax declaration, SemanticModel semanticModel)
         {
-            var classSymbol = semanticModel.GetDeclaredSymbol(classDeclaration);
+            var symbol = semanticModel.GetDeclaredSymbol(declaration);
 
-            return classSymbol != null &&
-                !classSymbol.IsSealed &&
-                !classSymbol.IsStatic &&
```

Comment:
This check has disappeared in the resulting code. Is that somewhat expected?

Reply:
Yes, this branch was never true since we check first for explicit interface specifiers and these cannot be present for static types (static types cannot implement interfaces).